src/core/account/account.ts

```diff
@@ -0,0 +1,252 @@
+  privateKey: Ed25519PrivateKey;
+  address?: AccountAddressInput;
+  legacy?: true;
+}
+
+/**
+ * Arguments for creating an `SingleKeyAccount` from an `Ed25519PrivateKey`.
+ * The `legacy` argument needs to be explicitly set to false in order to
+ * use the `SingleKey` authentication scheme.
+ */
+export interface CreateEd25519SingleKeyAccountFromPrivateKeyArgs {
+  privateKey: Ed25519PrivateKey;
+  address?: AccountAddressInput;
+  legacy: false;
+}
+
+/**
+ * Arguments for creating an `SingleKeyAccount` from any supported private key
+ * that is not an `Ed25519PrivateKey`.
+ * The `legacy` argument defaults to false and cannot be explicitly set to true.
+ */
+export interface CreateSingleKeyAccountFromPrivateKeyArgs {
+  privateKey: Exclude<PrivateKey, Ed25519PrivateKey>;
+  address?: AccountAddressInput;
+  legacy?: false;
+}
+
+/**
+ * Arguments for creating an opaque `Account` from any supported private key.
+ * This is used when the private key type is not known at compilation time.
+ */
+export interface CreateAccountFromPrivateKeyArgs {
+  privateKey: PrivateKey;
+  address?: AccountAddressInput;
+  legacy?: boolean;
+}
+
+/**
+ * Arguments for generating an `Ed25519Account`.
+ * This is the input type used by default.
+ */
+export interface GenerateEd25519AccountArgs {
+  scheme?: SigningSchemeInput.Ed25519;
+  legacy?: true;
+}
+
+/**
+ * Arguments for generating an `SingleKeyAccount` with ah underlying `Ed25519PrivateKey`.
+ * The `legacy` argument needs to be explicitly set to false,
+ * otherwise an `Ed25519Account` will be returned instead.
+ */
+export interface GenerateEd25519SingleKeyAccountArgs {
+  scheme?: SigningSchemeInput.Ed25519;
+  legacy: false;
+}
+
+/**
+ * Arguments for generating an `SingleKeyAccount` with any supported private key
+ * that is not an `Ed25519PrivateKey`.
+ * The `legacy` argument defaults to false and cannot be explicitly set to true.
+ */
+export interface GenerateSingleKeyAccountArgs {
+  scheme: Exclude<SigningSchemeInput, SigningSchemeInput.Ed25519>;
+  legacy?: false;
+}
+
+/**
+ * Arguments for generating an opaque `Account`.
+ * This is used when the input signature scheme is not known at compilation time.
+ */
+export interface GenerateAccountArgs {
+  scheme?: SigningSchemeInput;
+  legacy?: boolean;
+}
+
+/**
+ * Arguments for deriving a private key from a mnemonic phrase and a BIP44 path.
+ */
+export interface PrivateKeyFromDerivationPathArgs {
+  path: string;
+  mnemonic: string;
+}
+
+/**
+ * Interface for a generic Aptos account.
+ *
+ * The interface is defined as abstract class to provide a single entrypoint for account generation,
+ * either through `Account.generate()` or `Account.fromDerivationPath`.
+ * Despite this being an abstract class, it should be treated as an interface and enforced using
+ * the `implements` keyword.
+ *
+ * Note: Generating an account instance does not create the account on-chain.
+ */
+export abstract class Account {
```

Comment:
As part of the new wallet standard, we had the AccountInfo interface to hold an account of type Account
```
export interface AccountInfo = { account: Account, ansName?: string }
```
this was mostly done so it would be easier for us to resolve the account type (single signer/ multi signer) using the `variant` field - was this changed with this PR?

Reply:
Also, where did `AptosAccountVariant` enum go?

Reply:
I got some feedback regarding having both `Signer` and `Account` from @heliuchuan .
The argument is that it would be a breaking change, and the `Signer` name wouldn't be as intuitive.

So in this PR we only have `Account` and it has signing capabilities.
I think that for `AccountInfo` we can have the following struct, which would be functionally equivalent and similar to what we had before:

```
export interface AccountInfo = { address: AccountAddress, publicKey: AccountPublicKey, ansName?: string }
```
would also be nice to make it BCS serializable.

Reply:
Let's update the standard AIP then. The AIP currently relies on that change we planned to have